### ISSUE TITLE
Investigate perf issues

### ISSUE THREAD

**Issue by madeleineostoja** [SimplaElements/simpla-paths#13]
Very unscientiffic test, but ran Fetch as Google on simplajs.org, and the only thing that rendered was the one bit that _isn't_ using simpla-paths (the footer)

<img width="546" alt="screen shot 2017-12-16 at 4 26 36 pm" src="https://user-images.githubusercontent.com/6374876/34067584-f2eb8cd2-e27d-11e7-899c-29c9311b292d.png">

Might be worth profiling, and maybe even suggesting to deprecate this in favour of build-time path construction. Or, add a prerender buildstep to simpla-paths, so you can maintain easy clientside path recalculation

**Comment by bedeoverend** [comment]
Hmm...that is strange. I might try a version which is path-less, see if that is definitely what causes the issues. Regardless though, I'd be happy to deprecate, let people use frameworks / build time / backend to implement if they want it. In my experience, I've seen a lot of people doing their own namespacing, or none at all.

**Comment by madeleineostoja** [comment]
Yeah, I think the biggest advantage of `simpla-paths` though is how easy it is to dynamically change path trees on the client - eg: just wrap stuff in an `en` gid and boom global localisation. That would be nice to keep.

**Comment by bedeoverend** [comment]
Yeah - it's a nice feature. Perhaps we just should just add some disclaimers, less a core part to be used. To be fair I still haven't implemented that use case, even though I like it in theory.

Also having them as paths either statically or in a build step would make things like SSR a lot easier as you can do easy static analysis on the HTML

**Comment by madeleineostoja** [comment]
maybe this could somehow become an additive thing? Something you bolt on to existing paths just to dynamically modify them somewhere in the tree? Not sure what that would look like though
